Lib/socket.py

```diff
@@ -650,6 +650,23 @@ def socketpair(family=AF_INET, type=SOCK_STREAM, proto=0):
                 raise
         finally:
             lsock.close()
+
+        # Authenticating avoids using a connection from something else
+        # able to connect to {host}:{port} instead of us.
+        # We expect only AF_INET and AF_INET6 families.
+        try:
+            if (
+                ssock.getsockname() != csock.getpeername()
+                or csock.getsockname() != ssock.getpeername()
+            ):
+                raise ConnectionError("Unexpected peer connection")
```

Comment:
Is it possible to repeatedly call accept() until we find the legitimate socket call?

Reply:
Then malicious processes could just make us busy loop forever if the OS happens to pick their connections first. Looping "a few" times would be fine, but can always be DoSed anyways so why bother? Looping retrying forever seems unwise as it's be wasteful and callers expect this call to so very little work and immediately return. At some point the error being raised is important because it's a signal to the process that something is going wrong on the system.

Reply:
> Then malicious processes could just make us busy loop forever

But that's only achievable by the malicious process also busy looping with connection attempts, and it's quite unlikely to win all of them forever.

Reply:
I agree it's not worth adding a looping retry as part of this issue resolution